src/transformers/generation_flax_utils.py

```diff
@@ -101,12 +101,56 @@ def _run_loop_in_debug(cond_fn, body_fn, init_state):
             state = body_fn(state)
         return state
 
+    def _prepare_encoder_decoder_kwargs_for_generation(self, input_ids, model_kwargs):
+        encoder_kwargs = {
+            argument: value
+            for argument, value in model_kwargs.items()
+            if not (argument.startswith("decoder_") or argument.startswith("cross_attn"))
+        }
+        model_kwargs["encoder_outputs"] = self.encoder(input_ids, return_dict=True, **encoder_kwargs)
+        return model_kwargs
+
+    def _prepare_decoder_input_ids_for_generation(
+        self, input_ids, decoder_start_token_id: int = None, bos_token_id: int = None
+    ):
+        decoder_start_token_id = self._get_decoder_start_token_id(decoder_start_token_id, bos_token_id)
+        decoder_input_ids = jnp.ones((input_ids.shape[0], 1), dtype="i4") * decoder_start_token_id
+        return decoder_input_ids
+
+    def _get_decoder_start_token_id(self, decoder_start_token_id: int = None, bos_token_id: int = None) -> int:
```

Comment:
actually to begin with can we make this super simple and just use `self.config.decoder_start_token_id`? If the model doesn't have a `decoder_start_token_id` then we'll raise an error. Don't want to introduce all those exceptions we have in PT here in the beginning as well

Reply:
They are mostly for the EncoderDecoder class BTW